app/microwave-map.tsx

```diff
@@ -0,0 +1,80 @@
+import * as Haptics from 'expo-haptics';
+import * as Linking from 'expo-linking';
+import { Stack } from 'expo-router';
+import { View, Text, Platform } from 'react-native';
+import MapView, { Marker, Callout, CalloutSubview } from 'react-native-maps';
+
+import { Container } from '~/components/Container';
+import TopBar from '~/components/TopBar';
+import { MICROWAVE_LOCATION } from '~/data/MicrowaveLocations';
+import { COLORS } from '~/utils/colors';
+
+const MapMarkers = () => {
+  return (
+    <>
+      {MICROWAVE_LOCATION.map((location, index) => (
+        <Marker
+          key={index}
+          coordinate={location.coordinates}
+          title={location.name}
+          pinColor={COLORS['ut-burnt-orange']}>
+          <Callout>
+            <View className="p-2">
+              <CalloutSubview
+                className="my-10"
+                onPress={() => {
+                  Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Light);
+                  const encodedAddress = encodeURIComponent(location.address.replace(/\s/g, '+'));
+                  const url =
+                    Platform.OS === 'ios'
+                      ? `maps://maps.apple.com/?address=${encodedAddress}`
+                      : `https://www.google.com/maps/place/${encodedAddress}`;
+                  Linking.openURL(url);
```

Comment:
nit. I would extract this into a dedicated function to make the code more organized, clear, and concise.